barrier.go

```diff
@@ -80,7 +80,7 @@ func (n *BarrierNode) newBarrier(group edge.GroupInfo, first edge.PointMeta) (ed
 		)
 		return periodicBarrier, periodicBarrier.Stop, nil
 	default:
-		return nil, nil, errors.New("unreachable code, barrier node should have non-zero idle or non-zero period")
+		return nil, nil, errors.New("unreachable code, barrier node should have positive idle or positive period")
```

Comment:
I'm curious, what is the behavior supposed to be if `period` and `idle` are both set? Is that possible?

Reply:
The docs say it's based on one of them. I wonder if you should return an error if both are set?

Reply:
That is probably a good idea.  I think it is outside the scope of this PR though.

Reply:
Yeah, that makes sense.